clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -188,13 +188,15 @@ private boolean changeSubscription(Set<String> topicsToSubscribe) {
     /**
      * Add topics to the current group subscription. This is used by the group leader to ensure
      * that it receives metadata updates for all topics that the group is interested in.
+     *
      * @param topics The topics to add to the group subscription
+     * @return true if the group subscription contains topics which are not part of the local subscription
      */
     synchronized boolean groupSubscribe(Collection<String> topics) {
         if (!hasAutoAssignedPartitions())
             throw new IllegalStateException(SUBSCRIPTION_EXCEPTION_MESSAGE);
-        groupSubscription = new HashSet<>(groupSubscription);
-        return groupSubscription.addAll(topics);
+        groupSubscription = new HashSet<>(topics);
```

Comment:
We were accumulating topics earlier, but now we are replacing groupSubscription. I guess that is fine since we are passing in all the topics. May be worth updating the comment and also verifying that we are no longer accumulating in the unit test.